### DISCUSSION TITLE
How to encrypt application image using imgtool

### DISCUSSION THREAD

**Discussion by Sathishkumar-rk** [zephyrproject-rtos/zephyr#46792] [Q&A]
Hi Team,

I want to use encryption for primary slot image, if there any supporting document for that? i was referring below document (https://docs.zephyrproject.org/3.0.0/reference/kconfig/CONFIG_MCUBOOT_ENCRYPTION_KEY_FILE.html) but couldn't able to build encrypted image.

with the above documentation i was tried to sign and encrypt the image file, getting output binary as signed not encrypted. Find the imgtool footprint for reference.

west sign -t imgtool -- --key .\bootloader\mcuboot\enc-rsa2048-priv.pem --encrypt .\bootloader\mcuboot\enc-rsa2048-pub.pem --encrypt-keylen 128
=== image configuration:
partition offset: 131072 (0x20000)
partition size: 3145728 (0x300000)
rom start offset: 1024 (0x400)
=== signing binaries
unsigned bin: D:\work\Zephyr\zephyrproject\build\zephyr\zephyr.bin
signed bin:   D:\work\Zephyr\zephyrproject\build\zephyr\zephyr.signed.bin
unsigned hex: D:\work\Zephyr\zephyrproject\build\zephyr\zephyr.hex
signed hex:   D:\work\Zephyr\zephyrproject\build\zephyr\zephyr.signed.hex

can you guys help on this? what i'm doing wrong here? Let me know if any further information is required.

Regards,
Sathishkumar K

**Comment by DJBlom** [comment]
Did you solve this problem? I am stuck with the same issue. The only difference is that I am giving the encryption and the signing to west instead of using imgtool directly
